### PR TITLE
SITES-676: Prevent duplicate values when creating new user.

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Fixes issue with creating a new user.

# Needed By (Date)
- Soon as you can

# Urgency
- 7/10

# Steps to Test

1. Spin up a site on the 7.x-2.x branch of this module
2. Log in and try to create a local user and a SSO user. Get error on both.
3. Check out this branch.
4. Repeat step 2 but with success

# Associated Issues and/or People
- SITES-676

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: stanford_ssp.module [hunk 1]
```diff
@@ -365,13 +365,19 @@ function stanford_ssp_user_insert(&$edit, $account, $category) {
...
+
   // If the new account and the current user don't have matching emails then
   // the account must have been created by a form or drush instead of through
   // the user actually doing a login themselves.
   if (isset($attributes['email'][0]) && ($attributes['email'][0] !== $account->init)) {
     return;
   }
 
+  // Different SPs have different values for this...
+  if (isset($attributes['mail'][0]) && ($attributes['mail'][0] !== $account->init)) {
```
[the remark this answers is not in the document]
Comment (sherakama) [reply stanford_ssp.module]: This is not an object. I confirmed that this is indeed an array.

**Comment by jbickar** [comment]
So does this only happen when the user's canonical SUNetID is different from their email address? That would 'splain a lot.

**Comment by jbickar** [comment]
Also, if that's the case, I cannot test this. But I know someone who can...

**Comment by sherakama** [comment]
No, this has to do with the response from the SAML request. In the response that I had previously used the key for the email address was `email` but in the new set up it is simply `mail`. The validation code was not able to do an accurate match as the key was missing. Having a different email from sunet shouldn't affect this but is worth testing to be sure.

**Comment by jbickar** [comment]
OK, I'll have to look closer to see why it was getting a duplicate user error for the _active_ `$user`, not the user being added.

**Comment by sherakama** [comment]
In order to have the 'auto register' functionality working we use hook_user_insert() which, on any user creation, checks to see if it needs to be a SAML user. This code was not able to identify that the current user was not the new user as the key in the SAML attributes array was different.

**Pull request merged**